src/esn.calendar.libs/app/components/event-recurrence-edition/event-recurrence-edition.spec.js

```diff
@@ -45,115 +51,125 @@ describe('The event-recurrence-edition component', function() {
+      }).then(done);
     });
 
-    it('should push the weekday and sort the array', function() {
+    it('should push the weekday and sort the array', function(done) {
       this.$scope.event.rrule = {
         freq: undefined,
         interval: null
       };
-      this.initDirective(this.$scope);
-      this.eleScope.vm.event.rrule.byday = ['SU', 'WE', 'TU', 'MO'];
-      this.eleScope.vm.toggleWeekdays('FR');
-      expect(this.eleScope.vm.event.rrule.byday).to.deep.equal(['MO', 'TU', 'WE', 'FR', 'SU']);
+      this.initDirective(this.$scope).then(function() {
+        self.eleScope.vm.event.rrule.byday = ['SU', 'WE', 'TU', 'MO'];
+        self.eleScope.vm.toggleWeekdays('FR');
+        expect(self.eleScope.vm.event.rrule.byday).to.deep.equal(['MO', 'TU', 'WE', 'FR', 'SU']);
+      }).then(done);
     });
   });
 
   describe('at end date min value', function() {
-    it('should be today', function() {
+    it('should be today', function(done) {
       this.$scope.event.start = calMoment('2017-09-11 09:30');
       this.$scope.event.rrule = {
         freq: 'WEEKLY'
       };
-      this.initDirective(this.$scope);
-      var calMinDateAsString = this.eleScope.vm.getMinDate();
+      this.initDirective(this.$scope).then(function() {
+        var calMinDateAsString = self.eleScope.vm.getMinDate();
 
-      expect(calMinDateAsString).to.be.equal(calNow.format('YYYY-MM-DD'));
+        expect(calMinDateAsString).to.be.equal(calNow.format('YYYY-MM-DD'));
+      }).then(done);
     });
 
-    it('should be the event start date', function() {
+    it('should be the event start date', function(done) {
       this.$scope.event.start = calMoment().add(7, 'days');
       this.$scope.event.rrule = {
         freq: 'WEEKLY'
       };
-      this.initDirective(this.$scope);
-      var calMinDateAsString = this.eleScope.vm.getMinDate();
+      this.initDirective(this.$scope).then(function() {
+        var calMinDateAsString = self.eleScope.vm.getMinDate();
 
-      expect(calMinDateAsString).to.be.equal(this.$scope.event.start.format('YYYY-MM-DD'));
+        expect(calMinDateAsString).to.be.equal(self.$scope.event.start.format('YYYY-MM-DD'));
+      }).then(done);
     });
   });
 
   describe('scope.selectEndRadioButton', function() {
-    it('should set the correct radio button to checked', function() {
+    it('should set the correct radio button to checked', function(done) {
       this.$scope.event.rrule = {
         freq: 'WEEKLY'
       };
-      var element = this.initDirective(this.$scope);
-
-      this.eleScope.selectEndRadioButton(2);
-      var radio = angular.element(element).find('input[name="inlineRadioEndOptions"]')[2];
-
-      expect(radio.checked).to.be.true;
+      this.initDirective(this.$scope).then(function(element) {
+        self.eleScope.selectEndRadioButton(2);
+        var radio = angular.element(element).find('input[name="inlineRadioEndOptions"]')[2];
+  
+        expect(radio.checked).to.be.true;
+      }).then(done)
```

Comment:
```suggestion
        expect(radio.checked).to.be.true;
        done();
      })
```

Reply:
Updated. Writing another then to make it easier to write catch also. Please see the comment below from @tuanlc , I have resolved that.